crates/matrix-sdk/src/media.rs

```diff
@@ -82,6 +82,37 @@ impl MediaFileHandle {
     pub fn path(&self) -> &Path {
         self.file.path()
     }
+
+    /// Persist the media file to the given path.
+    pub fn persist(self, path: &Path) -> Result<File, PersistError> {
+        self.file.persist(path).map_err(|e| PersistError {
+            error: e.error,
+            file: Self { file: e.file, _directory: self._directory },
+        })
+    }
+}
+
+/// Error returned when [`MediaFileHandle::persist`] fails.
+#[cfg(not(target_arch = "wasm32"))]
+pub struct PersistError {
+    /// The underlying IO error.
+    pub error: io::Error,
+    /// The temporary file that couldn't be persisted.
+    pub file: MediaFileHandle,
+}
+
+#[cfg(not(any(target_arch = "wasm32", tarpaulin_include)))]
```

Comment:
What are all these cfgs about?
I suspect it's not to include such code in the wasm builds, but why?

Reply:
This means "only include this code if the conditions are valid", and the condition means "we're not on the wasm platform or using tarpaulin for code coverage testing".

The `MediaFileHandle` isn't available on wasm in the first place, and custom debug implementations require the tarpaulin annotation to not count as code not covered during code coverage tracking.

Reply:
Why `MediaFileHandle` is not available on webasm?